pkg/provisioner/controllerprovisioner.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioner
+
+import (
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/azureutils"
+)
+
+func NewControllerProvisioner(useAzureStorage bool, kubeConfig *rest.Config, kubeClient clientset.Interface) (*ControllerProvisioner, error) {
+	if useAzureStorage {
```

Comment:
We could omit `useAzureStorage` and the relevant command-line flag for now since it really has no effect. I suspect we would want to use an enumerate type here though to allow for more than just two.

Reply:
Yes, I agree with you. I also want an enum to be used. I have placed the flag as a placeholder, incase we have a clarity in the PR discussion.